src/components/Editor/tests/DebugLine.spec.js

```diff
@@ -0,0 +1,132 @@
+/* eslint max-nested-callbacks: ["error", 7] */
+
+import React from "react";
+import { shallow } from "enzyme";
+
+import DebugLine from "../DebugLine";
+
+import { makeSourceRecord } from "../../../utils/test-head";
+
+import { setDocument } from "../../../utils/editor/source-documents";
+
+function createMockDocument(clear) {
+  const doc = {
+    addLineClass: jest.fn(),
+    removeLineClass: jest.fn(),
+    markText: jest.genMockFunction().mockReturnValue({ clear })
+  };
+
+  return doc;
+}
+
+function generateDefaults(editor, overrides) {
+  return {
+    editor,
+    pauseInfo: {
+      why: { type: "breakpoint" }
+    },
+    selectedFrame: null,
+    selectedSource: makeSourceRecord("foo"),
+    ...overrides
+  };
+}
+
+function createFrame(line) {
+  return {
+    location: {
+      sourceId: "foo",
+      line,
+      column: 2
+    }
+  };
+}
+
+function render(overrides = {}) {
+  const clear = jest.fn();
+  const editor = { codeMirror: {} };
+  const props = generateDefaults(editor, overrides);
+
+  const doc = createMockDocument(clear);
+  setDocument(props.selectedSource.get("id"), doc);
+
+  const component = shallow(<DebugLine.WrappedComponent {...props} />, {
+    lifecycleExperimental: true
+  });
+  return { component, props, clear, editor, doc };
+}
+
+describe("DebugLine Component", () => {
+  describe("pausing at the first location", () => {
+    it("should show a new debug line", async () => {
+      const { component, props, doc } = render({
+        selectedSource: makeSourceRecord("foo", { loadedState: "loaded" })
+      });
+
+      const selectedFrame = createFrame(2);
```

Comment:
the `2` here might be better in a const. that way the relationship between it and the expected result is clear and descriptive. 

```javascript
const lineNumber = 2;
const editorNumber = lineNumber - 1;
createFrame(lineNumber)

//.. 
expect(doc.addLineClass.mock.calls).toEqual([
  [editorLine,  "line", "new-debug-line"]
]);